### ISSUE TITLE
Generated Error and Recover do not pass the context

### ISSUE THREAD

**Issue by rhymes** [vektah/gqlgen#103]
### Expected Behaviour

According to https://github.com/vektah/gqlgen/blob/ab3803a6683be2941e5817ccb9f34460f2b39ac9/graphql/recovery.go#L11 

the recovery function should be called with:

```go
ec.Recover(ctx, r)
```

Same for the Error function https://github.com/vektah/gqlgen/blob/0c7bdfc682bc065ff44752ac4fbefd455c3fcfeb/graphql/error.go#L44

unfortunately in my case they are generated without the context as the first parameter which means that the binary won't build

### Actual Behavior

Instead both functions should be generated with the context as the first parameter.

The weird thing is that in the template they both have the context as the first parameter: https://github.com/vektah/gqlgen/blob/master/codegen/templates/field.gotpl

### Minimal graphql.schema and models to reproduce

I have only two instances where it happen, and they are both query methods, example:

```go
func (ec *executionContext) _Query_app(ctx context.Context, field graphql.CollectedField) graphql.Marshaler {
	var arg0 string
	if tmp, ok := field.Args["appID"]; ok {
		var err error
		arg0, err = graphql.UnmarshalID(tmp)
		if err != nil {
			ec.Error(err)
			return graphql.Null
		}
	}
	return graphql.Defer(func() (ret graphql.Marshaler) {
		defer func() {
			if r := recover(); r != nil {
				userErr := ec.Recover(r)
				ec.Error(userErr)
				ret = graphql.Null
			}
		}()
		rctx := graphql.WithResolverContext(ctx, &graphql.ResolverContext{Field: field})
		res, err := ec.resolvers.Query_app(rctx, arg0)
		if err != nil {
			ec.Error(err)
			return graphql.Null
		}
		if res == nil {
			return graphql.Null
		}
		return ec._App(ctx, field.Selections, res)
	})
}
```

As you can see both Recover and Error are generated without the context.

I'm using revision 263040ce1a362270b5897a5982572ddc1fe807be

Any ideas?

**Comment by vektah** [comment]
I think you have a desync between the version of gqlgen you are using to generate and the one you have vendored in.

There are a bunch of approaches to pinning binary versions, I think the simplest is https://github.com/vektah/gorunpkg.

There's a little bit of detail in https://gqlgen.com/getting-started/#finishing-touches

**Comment by rhymes** [comment]
@vektah probably but I'm already using gorunpkg like in the docs I guess

This is the content of my file generation file:

```go
//go:generate gorunpkg github.com/vektah/gqlgen -typemap types.json -package gqlapp
package gqlapp
```

and this is the content of the Gopkg.lock

```text
[[projects]]
  branch = "master"
  name = "github.com/vektah/gqlgen"
  packages = [
    "graphql",
    "handler",
    "neelance/common",
    "neelance/errors",
    "neelance/introspection",
    "neelance/query",
    "neelance/schema",
    "neelance/validation"
  ]
  revision = "2a437c23b379cdcad7b7f4ca2e14a5c6075123a9"
```

**Comment by vektah** [comment]
Did you ever get to the bottom of this? The generated code you posted definitely cant be generated on master.

**Comment by rhymes** [comment]
@vektah I did the following:

* went in $GOPATH
* deleted the folder `github.com/vektah/gorunpkg` (also noticed I had a global gqlgen as well, deleted that too)
* issued `go get github.com/vektah/gorunpkg`
* went back to my project
* removed the vendor folder
* issued `dep ensure` (I have gqlgen as a dependency in the toml)
* issued `go generate ./...`
* got an error because it cannot find gqlgen:

```text
go generate ./...
pkg: cannot find package "github.com/vektah/gqlgen" in any of:
	/.../vendor/github.com/vektah/gqlgen (vendor tree)
	/usr/local/Cellar/go/1.10.2/libexec/src/github.com/vektah/gqlgen (from $GOROOT)
	/.../go/src/github.com/vektah/gqlgen (from $GOPATH)
```
 * reinstalled gqlgen globally as well `go get github.com/vektah/gqlgen`
 * reissued `go generate ./...`
 * had with this issue on build:

```text
generated.go:69:25: undefined: graphql.ErrorResponse
generated.go:132:6: rctx.PushField undefined (type *graphql.ResolverContext has no field or method PushField)
generated.go:133:12: rctx.Pop undefined (type *graphql.ResolverContext has no field or method Pop)
```

(this is probably because the global gqlgen and the one in the vendor are not the same version)

* updated the vendored gqlgen (to revision `6c823beb069be4fed6f0218ec5ff6a5211968b56`)
* re-built the project
* finally worked

Basically I need to make sure that the global gqlgen (used to generate) and the gqlgen embedded in the app (used build the binary) are aligned.

Is there a way to use the vendored version to generate as well?

Now my generation comment is the following:

```go
//go:generate gorunpkg github.com/vektah/gqlgen -typemap types.json -package backend
```

and it runs the global gqlgen in $GOPATH, not the vendored one by dep

**Comment by vektah** [comment]
what version of go are you running?

**Comment by rhymes** [comment]
`go version go1.10.2 darwin/amd64`

**Comment by vektah** [comment]
Are you on the gophers slack? lets do some debugging.

gophers.slack.com @vektah

**Comment by rhymes** [comment]
But wait, not it works.

It makes sense that the global gqlgen version and the one embedded need to be aligned. 

I just need to remember that :)

**Comment by vektah** [comment]
My gut feeling is that gorunpkg is out of date or misbehaving on your system for some reason. Did you `go get -u` the U is important for updating otherwise it just rebuilds whatever you have locally.

**Comment by rhymes** [comment]
I solved it!

It was my Gopkg.toml file.

I added gqlgen as a normal dependency, instead it has to be in the "required" part.

Instead of:

```ini

[[constraint]]
  branch = "master"
  name = "github.com/vektah/gqlgen"
```

it should be

```ini
required = ["github.com/vektah/gqlgen"]
```

I removed the global gqlgen, re-generate and it all works :)

**Comment by vektah** [comment]
Glad you got it solved :+1:
